File: 필수문항/169_추월.py
import sys
from collections import deque

input = sys.stdin.readline

n = int(input())
in_list = [input().rstrip() for _ in range(n)]
out_list = deque([input().rstrip() for _ in range(n)])
ans_list = []
now = 0

while out_list:
    if in_list[now] in ans_list:
        now += 1
        continue
    out = out_list.popleft()
    if out not in ans_list:
        if out != in_list[now]:
            ans_list.append(out)
        else:
            now += 1

print(len(ans_list))

File: 필수문항/169_추월.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String[] inList = new String[n];
        String[] outList = new String[n];
        for (int i = 0; i < n; i++) {
            inList[i] = br.readLine();
        }
        for (int i = 0; i < n; i++) {
            outList[i] = br.readLine();
        }
        int now = -1;
        int ans = 0;
        for (int i = 0; i < n; i++) {
            for (int j = now + 1; j < n; j++) {
                if (inList[i].equals(outList[j])) {
                    ans++;
                    now = j;
                    break;
                }
            }
        }
        System.out.println(n - ans);
    }
}
